dad is crashing when `pip` missing from bin/

__Description__

My virtualenv was badly initialized, and dad crashed when trying to pip install the requirements

Step to reproduce:

1. Create a python dad project
2. dad up
3. remove pip from bin folder of the corresponding virtualenv
4. dad up again
```
➜  dataproc-test git:(master) ✗ dad up
◼︎ Unknown task (gcloud)
◼︎ Python (3.6.5)
  Already OK!
◼︎ Pip (requirements.txt)
  Running: /Users/mlhamel/.local/share/dad/virtualenvs/dataproc-test-4030403424-3.6.5/bin/pip install -r requirements.txt
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x118fe1c]

goroutine 1 [running]:
syscall.syscall.WaitStatus.ExitStatus(...)
	/home/travis/.gimme/versions/go1.9.4.linux.amd64/src/os/exec_posix.go:77
github.com/pior/dad/pkg/executor.GetExitCode(...)
	/home/travis/gopath/src/github.com/pior/dad/pkg/executor/executor.go:20
github.com/pior/dad/pkg/executor.RunSilent(0xc4200f6000, 0x52, 0xc420041bc0, 0x3, 0x3, 0x0, 0x1b, 0x1)
	/home/travis/gopath/src/github.com/pior/dad/pkg/executor/executor.go:41 +0x14c
github.com/pior/dad/pkg/executor.Run(0xc4200f6000, 0x52, 0xc420041bc0, 0x3, 0x3, 0xc420014801, 0x10, 0xc4200e0060)
	/home/travis/gopath/src/github.com/pior/dad/pkg/executor/executor.go:26 +0x343
github.com/pior/dad/pkg/tasks.(*Pip).Perform(0xc42000a880, 0xc42000a8a0, 0x0, 0x0)
	/home/travis/gopath/src/github.com/pior/dad/pkg/tasks/pip.go:58 +0x1ff
github.com/pior/dad/pkg/tasks.RunAll(0xc420054300, 0xc420084460, 0xc42004a950, 0x120c4de, 0x4)
	/home/travis/gopath/src/github.com/pior/dad/pkg/tasks/runner.go:27 +0x121
github.com/pior/dad/pkg/cmd.upRun(0x132ccc0, 0x134cdc0, 0x0, 0x0)
	/home/travis/gopath/src/github.com/pior/dad/pkg/cmd/up.go:28 +0xdc
github.com/pior/dad/vendor/github.com/spf13/cobra.(*Command).execute(0x132ccc0, 0x134cdc0, 0x0, 0x0, 0x132ccc0, 0x134cdc0)
	/home/travis/gopath/src/github.com/pior/dad/vendor/github.com/spf13/cobra/command.go:750 +0x2c1
github.com/pior/dad/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc42008c240, 0x1000100, 0x0, 0xffffffffffffffff)
	/home/travis/gopath/src/github.com/pior/dad/vendor/github.com/spf13/cobra/command.go:831 +0x30e
github.com/pior/dad/vendor/github.com/spf13/cobra.(*Command).Execute(0xc42008c240, 0x25, 0x119d6b7)
	/home/travis/gopath/src/github.com/pior/dad/vendor/github.com/spf13/cobra/command.go:784 +0x2b
github.com/pior/dad/pkg/cmd.Execute(0x12343c0, 0x25)
	/home/travis/gopath/src/github.com/pior/dad/pkg/cmd/root.go:58 +0x4a
main.main()
	/home/travis/gopath/src/github.com/pior/dad/dad.go:10 +0x39
```

--- a/pkg/executor/executor.go
+++ b/pkg/executor/executor.go
@@ -27,14 +27,18 @@ func (e *Executor) SetCwd(cwd string) *Executor {
 }
 
 func (e *Executor) getExitCode(err error) (int, error) {
-	if err != nil {
-		if exitError, ok := err.(*exec.ExitError); ok {
-			code := exitError.Sys().(syscall.WaitStatus).ExitStatus()
-			return code, nil
-		}
+	if err == nil {
+		code := e.cmd.ProcessState.Sys().(syscall.WaitStatus).ExitStatus()
+		return code, nil
 	}
-	code := e.cmd.ProcessState.Sys().(syscall.WaitStatus).ExitStatus()
-	return code, err
+
+	if exitError, ok := err.(*exec.ExitError); ok {
+		code := exitError.Sys().(syscall.WaitStatus).ExitStatus()
+		return code, nil
+	}
+
+	// There was an error but not a ExitError, just return it with an invalid exit code
+	return -1, err
 }
 
 // Run executes the command and returns the exit code

--- a/pkg/executor/executor_test.go
+++ b/pkg/executor/executor_test.go
@@ -49,3 +49,10 @@ func TestShellCapturePWD(t *testing.T) {
 	require.Zero(t, code)
 	require.Equal(t, "/bin\n", output)
 }
+
+func TestCommandNotFound(t *testing.T) {
+	code, err := New("never-ever-cmd").Run()
+
+	require.Error(t, err)
+	require.Equal(t, -1, code)
+}

--- a/pkg/executor/legacy.go
+++ b/pkg/executor/legacy.go
@@ -11,14 +11,18 @@ import (
 )
 
 func getExitCode(err error, cmd *exec.Cmd) (int, error) {
-	if err != nil {
-		if exitError, ok := err.(*exec.ExitError); ok {
-			code := exitError.Sys().(syscall.WaitStatus).ExitStatus()
-			return code, nil
-		}
+	if err == nil {
+		code := cmd.ProcessState.Sys().(syscall.WaitStatus).ExitStatus()
+		return code, nil
 	}
-	code := cmd.ProcessState.Sys().(syscall.WaitStatus).ExitStatus()
-	return code, err
+
+	if exitError, ok := err.(*exec.ExitError); ok {
+		code := exitError.Sys().(syscall.WaitStatus).ExitStatus()
+		return code, nil
+	}
+
+	// There was an error but not a ExitError, just return it with an invalid exit code
+	return -1, err
 }
 
 func Run(program string, args ...string) (int, error) {

